src/OrchardCore/Orchard.Data/Migration/DataMigrationManager.cs

```diff
@@ -151,89 +152,87 @@ public class DataMigrationManager : IDataMigrationManager
             // apply update methods to each migration class for the module
             foreach (var migration in migrations)
             {
-                _session.ExecuteMigration(schemaBuilder =>
-                {
-                    migration.SchemaBuilder = schemaBuilder;
+                var schemaBuilder = new SchemaBuilder(_session);
```

Comment:
separation of concerns, this way the method doesn't have to be on ISession, which mean a lot less dependencies.